### ISSUE TITLE
False positive for `Style/ArgumentsForwarding` when using default argument values

### ISSUE THREAD

**Issue by vassyz** [rubocop/rubocop#12144]
I just updated to the latest `rubocop` version and was hoping that it would solve this false positive as I saw similar issues being reported. But it seems this is not the case.

--------

## Expected behavior

Leave the following code unchanged:

```ruby
def travel_temporarily_to(*datetime, freeze: true, &block)
  if freeze
    Timecop.freeze(*datetime, &block)
  else
    Timecop.travel(*datetime, &block)
  end
end
```

## Actual behavior

Formats to:

```ruby
def travel_temporarily_to(...)
  if freeze
    Timecop.freeze(...)
  else
    Timecop.travel(...)
  end
end
```

This means that `freeze` will not be set to `true` and instead be `nil`.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.56.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.1.2) [arm64-darwin22]
  - rubocop-capybara 2.18.0
  - rubocop-factory_bot 2.23.1
  - rubocop-rails 2.20.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.23.2
```

**Comment by owst** [comment]
Hi @vassyz sorry for the false positive. I think this is probably resolved by https://github.com/rubocop/rubocop/pull/12142 ?

**Comment by vassyz** [comment]
@owst No worries. Unfortunately, https://github.com/rubocop/rubocop/pull/12142 doesn't seem to be fixing this particular issue.

**Comment by vassyz** [comment]
@owst So sorry, I was being lazy. I tested it with formatting in VS Code and it didn't pick up the correct rubocop version.
Can confirm that the latest version from master fixes this issue. Thank you 🙇

**Comment by mildred** [comment]
I have an example where automatic correctioon of this rubocop rule produces erroneous code.

<details>
<summary>Original code</summary>

```rb
# frozen_string_literal: true

class CalculatorService
  attr_reader :value, :formula

  def initialize(formula, **vars)
    @error = false
    @formula = formula
    @formula_corrected = false

    @value = Dentaku::Calculator.new.evaluate(formula, **vars)

    if error?
      # Try formula by replacing `,` with decimal point `.`
      formula = formula.tr(',', '.').tr(';', ',')

      # Try formula by replacing `**` with `^`
      formula = formula.gsub('**', '^')

      @value = Dentaku::Calculator.new.evaluate(formula, **vars)

      unless error?
        @formula = formula
        @formula_corrected = true
      end
    end
  end

  def error?
    @value.nil? && formula.present?
  end

  def formula_corrected?
    @formula_corrected
  end
end
```

</details>

<details>
<summary>Reformatted code</summary>

```rb
# frozen_string_literal: true

class CalculatorService
  attr_reader :value, :formula

  def initialize(formula, **)
    @error = false
    @formula = formula
    @formula_corrected = false

    @value = Dentaku::Calculator.new.evaluate(formula, **vars)

    if error?
      # Try formula by replacing `,` with decimal point `.`
      formula = formula.tr(',', '.').tr(';', ',')

      # Try formula by replacing `**` with `^`
      formula = formula.gsub('**', '^')

      @value = Dentaku::Calculator.new.evaluate(formula, **)

      unless error?
        @formula = formula
        @formula_corrected = true
      end
    end
  end

  def error?
    @value.nil? && formula.present?
  end

  def formula_corrected?
    @formula_corrected
  end
end
```

</details>

The problem is that `**vars` is replaced by `**` only once but in fact it is used twice in the code. This produces a NameError: undefined local variable or method `vars`.

```
Offenses:

app/services/calculator_service.rb:6:27: C: [Corrected] Style/ArgumentsForwarding: Use anonymous keyword arguments forwarding (**).
  def initialize(formula, **vars)
                          ^^^^^^
app/services/calculator_service.rb:20:58: C: [Corrected] Style/ArgumentsForwarding: Use anonymous keyword arguments forwarding (**).
      @value = Dentaku::Calculator.new.evaluate(formula, **vars)
                                                         ^^^^^^

1 file inspected, 2 offenses detected, 2 offenses corrected
```

**Comment by owst** [comment]
Hi @mildred - apologies for the bug. The issue you're seeing seems related to but distinct from this issue (and it seems like a bug rather than a false-positive) Please can you raise another issue and include the various versions (Rubocop, Ruby via `rubocop -V`) that you're using and I'll take a look at a fix. Thanks

@bbatsov @koic this original issue can be closed please
